src/routes/controllers/articles.controller.js

```diff
@@ -41,14 +41,17 @@ export async function createArticle(req, res) {
     const userInfo = await verifyToken(
```

Comment:
Good job @igbominadeveloper so far, however, this function has been implemented as a middleware module named `verifyToken`, I suggest to use that function as a middleware in your route so that it can handle that verification for you.

Reply:
Thanks, @NedyUdombat for reviewing. I thought of it too but this route is supposed to be open to all and sundry. verifytoken middleware ensures that a valid token is set without which it does not allow the request through. So it couldn't be applied here.

Reply:
N.B verifyToken is a helper function not a middleware. The middleware is `checkAuthorizedUser`

You know the `checkAuthorizedUser` middleware throws you back if you don't provide a valid token right? It means if i apply it a public route like `get a single article`, unauthenticated users won't be able to read articles. That is why i didn't use it.

Reply:
Good job with the reading stats logic overall. If I understand what you have been explaining so far, you are allowing access to an article even if the user isn't authenticated. So this in-controller verification is for the users that are authenticated, right?

Reply:
@eben-k Checking if it is an authenticated user that is trying to read an article so I can include his userId while capturing his reading stats. if no token is set, the stat would just count as number of reads for the author of the article.